### PR TITLE
Implemented additional data for processing api

### PR DESCRIPTION
**Pull request by iovsn**
- Added viewAzimuthMean, sunAzimuthAngles, sunZenithAngles.
- Individual bands get downloaded per-datatype (uint8, uint16, float32), each in their own .tiff, but all in all request, compressed into a .tar file.
- They then get extracted and assigned to the appropriate features.

This is a WIP PR, to get some comments while I work on it. Some refactoring is still needed.

### CODE REVIEW

File: io/eolearn/io/processing_api.py [hunk 1]
```diff
@@ -283,43 +337,44 @@ def _request_payload(self, date_from, date_to, bbox, size_x, size_y):
...
 
-        images = ((img['default.tif'], img['userdata.json']) for img in images)
-        images = [(img, meta.get('norm_factor', 0)) for img, meta in images]
+        for rtype, tifs, bands in itr_tifs:
+            for band in bands:
+                feature = rtype.feature_type, band
+                eopatch[feature] = self._extract_array(tifs, bands.index(band), shape, rtype.np_dtype)
 
-        for f_type, f_name_src, f_name_dst in self.additional_data:
-            eopatch[(f_type, f_name_dst)] = self._extract_additional_data(images, f_type, f_name_src, shape)
```
Comment (mlubej) [inline io/eolearn/io/processing_api.py]: The new changes don't support the writing of the additional data under the new name of the feature.

**Comment by mlubej** [comment]
Checked for the problematic data of the sun angles. Now it seems to work as expected :+1: 

Before:

image

After:

image

File: io/eolearn/io/processing_api.py (hunk 1 above)
Comment (iovsn) [reply io/eolearn/io/processing_api.py]: Fixing it...

Comment (iovsn) [reply io/eolearn/io/processing_api.py]: Fixed.

**Comment by mlubej** [comment]
Is there anything else left to do for this branch?

**Comment by iovsn** [comment]
> Is there anything else left to do for this branch?

Just the minor test update I did and @AleksMat should probably also check it.

**Pull request merged**